pkg/addon/addon.go

```diff
@@ -952,16 +955,41 @@ func (h *Installer) installDependency(addon *InstallPackage) error {
 		if h.dryRun {
 			continue
 		}
-		// always install addon's latest version
-		depAddon, err := h.loadInstallPackage(dep.Name, dep.Version)
-		if err != nil {
-			return err
-		}
 		depHandler := *h
 		depHandler.args = nil
-		if err = depHandler.enableAddon(depAddon); err != nil {
-			return errors.Wrap(err, "fail to dispatch dependent addon resource")
+		var depAddon *InstallPackage
+		// try to install the dependent addon from the same registry with the current addon
+		depAddon, err = h.loadInstallPackage(dep.Name, dep.Version)
```

Comment:
Do we support specifying the registry of the dependency?

Reply:
Not yet.